analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/SecureRandomSeedsShouldNotBePredictable.cs

```diff
@@ -85,18 +95,26 @@ private static ProgramState ProcessArrayCreation(ProgramState state, IArrayCreat
     private static ProgramState ProcessArraySetValue(ProgramState state, IInvocationOperationWrapper invocation)
     {
         if (invocation.TargetMethod.Name == nameof(Array.SetValue)
-            && invocation.TargetMethod.ContainingType.Is(KnownType.System_Array))
+            && invocation.TargetMethod.ContainingType.Is(KnownType.System_Array)
+            && invocation.Instance.TrackedSymbol(state) is { } array)
         {
             return invocation.ArgumentValue("value") is { ConstantValue.HasValue: true }
                 ? state
-                : state.SetSymbolConstraint(invocation.Instance.TrackedSymbol(state), CryptographicSeedConstraint.Unpredictable);
+                : state.SetSymbolConstraint(array, CryptographicSeedConstraint.Unpredictable);
```

Comment:
This could result in NRE, since `invocation` might _not_ have an instance, for example when called inside the class it is defined, so you don't even need a `this` necessarily.